File: server/domain/dto/exporter.py
from typing import Optional

from pydantic import BaseModel

from server.domain.entity import ExportMode


class CreateExporter(BaseModel):
    title: str
    mode: ExportMode
    name_rule: Optional[str]


class UpdateExporter(BaseModel):
    id: int
    title: Optional[str]
    mode: Optional[ExportMode]
    name_rule: Optional[str]


class SelectExporter(BaseModel):
    id: Optional[int]
    title: Optional[str]

File: internal/model/dto/exporter.go
package dto

import "Scrapper/internal/model/entity"

type CreateExporter struct {
	Title    string
	Mode     entity.ExportMode
	NameRule string
}
type UpdateExporter struct {
	Id       uint
	Title    *string
	Mode     *entity.ExportMode
	NameRule *string
}
type SelectExporter struct {
	Id    uint
	Title string
}
